### ISSUE TITLE
2.2.1 - Strange Google assertion error

### ISSUE THREAD

**Issue by kalbermattenm** [camptocamp/ngeo#3237]
I just updated to GMF 2.2.1 and I get some kind of strange error (do not care about the Google API stuff). I do not really get what I am suppose to do to debug that...

Moreover in non-debug mode, I get more errors and the WMTS layers do not load.

![image](https://user-images.githubusercontent.com/1681332/33771518-8af01c1c-dc31-11e7-9838-09edde9720dc.png)

**Comment by fredj** [comment]
For the google API, if you don't use streetview you can safely remove the script tag in index.html (script tag with src=http://maps.google.com/maps )

**Comment by kalbermattenm** [comment]
Yes, I know, but I am using it (this is our Internet instance). The API key just cannot digest http://localhost:5000 as reference, but I think that this is independent from the `goog.assert` stuff which seems to be my blocker...

**Comment by kalbermattenm** [comment]
Just for info, this error actually appears once the layer tree is loaded (thus as soon as the "En chargement" message disappears).

**Comment by kalbermattenm** [comment]
And here is the content of the assertion error stack:
```
"AssertionError: Assertion failed
    at new goog.asserts.AssertionError (http://localhost:5000/node_modules/closure-util/.deps/library/0f95ad8150b77cd8db0e8d9e9726292ab33580af/closure/goog/asserts/asserts.js:62:20)
    at Object.goog.asserts.doAssertFailure_ (http://localhost:5000/node_modules/closure-util/.deps/library/0f95ad8150b77cd8db0e8d9e9726292ab33580af/closure/goog/asserts/asserts.js:121:11)
    at Object.goog.asserts.assert (http://localhost:5000/node_modules/closure-util/.deps/library/0f95ad8150b77cd8db0e8d9e9726292ab33580af/closure/goog/asserts/asserts.js:151:18)
    at gmfThemes_.getBgLayers.then (http://localhost:5000/node_modules/ngeo/contribs/gmf/src/controllers/abstract.js:606:18)
    at processQueue (http://localhost:5000/node_modules/angular/angular.js:17000:37)
    at http://localhost:5000/node_modules/angular/angular.js:17044:27
    at Scope.$digest (http://localhost:5000/node_modules/angular/angular.js:18182:15)
    at Scope.$apply (http://localhost:5000/node_modules/angular/angular.js:18480:24)
    at done (http://localhost:5000/node_modules/angular/angular.js:12501:47)
    at completeRequest (http://localhost:5000/node_modules/angular/angular.js:12727:7)"
```

**Comment by fredj** [comment]
Seems to be related to the background layer selection:
https://github.com/camptocamp/ngeo/blob/a0377296fb71ceb00911c2028262cf0dce337758/contribs/gmf/src/controllers/abstract.js#L606

**Comment by kalbermattenm** [comment]
`gmfThemes_.getBgLayers` refers to the layers inside the `background` group. This group is not used by sitn.  I created this (empty) group, but did not assign any layer to it and this triggers the bug !

I just added a layer to this group and I do not see this error any more.

Maybe, to avoid this error, it would be good to add an exception if a GMF instance does not use background layers... (we only use WMTS layers directly in the layer tree...)

By the way, many thanks to @rbovard 👍 🎉

**Comment by rbovard** [comment]
It's my Christmas present!

**Comment by kalbermattenm** [comment]
Yes, thanks !

But... Defining a layer in the background group makes this layer active, thus tiles for this "none" layer are loaded. It creates kind of a background image create by a layer which cannot be unactivated.

My solution is thus temporary...

**Comment by fredj** [comment]
@kalbermattenm this issue should have been fixed with #3374

Please re-open this issue if it's not the case

**Comment by kalbermattenm** [comment]
Oh, thanks ! I did not get that this PR was related with this issue.

I will try it out on Monday.

**Comment by kalbermattenm** [comment]
I am sorry @fredj but the issue seems not to be resolved. I applied your improvements (I pinged you in my PR in our project).

If you call the mobile app, you do not get any error (https://sitn.ne.ch/dev_internet2/mobile). If you call the debug, then you get the mentionned error (https://sitn.ne.ch/dev_internet2/mobile/?debug)

My config is not to declare anything for `functionnalities.basemap` (see project PR) and to define an empty `background` group. If this group is not empty, it calls the defined layer.

I do not know if I missed something... ?
